### ISSUE TITLE
nn.init.orthogonal segfaults

### ISSUE THREAD

**Issue by marikgoldstein** [pytorch/pytorch#8868]
## Issue description

Call to nn.init.orthogonal_ causes SEGFAULT on ubuntu and not on mac

## Code example

In the following code, the call that assigns `conv1` works fine but the one that assigns `conv2` causes a SEGFAULT and dumps core when it reaches `nn.init.orthogonal_(layer.weight.data)`.

This code works totally fine on a mac installation with the same pytorch library version (the newest). But this crashes on my Ubuntu (system info below). Any thoughts? Thanks!

```
def layer_init(layer, w_scale=1.0):
    nn.init.orthogonal_(layer.weight.data)
    layer.weight.data.mul_(w_scale)
    nn.init.constant_(layer.bias.data, 0)
    return layer

conv1 = layer_init(nn.Conv2d(4, 32, kernel_size=8, stride=4))
conv2 = layer_init(nn.Conv2d(32, 64, kernel_size=4, stride=2))
```
By the way, I came across this issue while I happened to be using this great pytorch DeepRL repo for recreating results from a lot of recent RL papers https://github.com/ShangtongZhang/DeepRL (code works well on my mac)

## System Info

PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: 8.0.61

OS: Ubuntu 16.04.4 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 7.5.17
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 375.66
cuDNN version: Probably one of the following:
/usr/local/lib/python2.7/dist-packages/torch/lib/libcudnn-3f9a723f.so.6.0.21

Versions of relevant libraries:
[pip3] numpy (1.13.1)
[pip3] torch (0.4.0)
[pip3] torchvision (0.1.8)
[conda] pytorch                   0.1.12          py36cuda8.0cudnn6.0_1  
[conda] torch                     0.4.0                     <pip>
[conda] torchvision               0.1.8                    py36_0

**Comment by marikgoldstein** [comment]
I tested the following in isolation of any other code:

```
import torch
import torch.nn

a = nn.Conv2d(4,32,kernel_size=8,stride=4)
b = nn.Conv2d(32,64,kernel_size=4,stride=2)
nn.init.orthogonal_(a.weight.data)
nn.init.orthogonal_(b.weight.data)
```

The first `orthogonal_` works and the second one segfaults.

**Comment by vishwakftw** [comment]
I am not able to reproduce on master. Could you try installing from source, and check if the problem persists?

**Comment by marikgoldstein** [comment]
I made a fresh ubuntu user account and did the minimal install necessary to get my python/pytorch environment setup. I installed pytorch with pip, not from source. 

The issue doesn't happen anymore! 

A couple of other people at school said this problem with orthogonal_ has happened to them too in the past. I'll close it for now but will see if I can track down the problem on the user account where it's broken.

Thanks!
